temporian/test/utils.py

```diff
@@ -80,12 +80,15 @@ def assertOperatorResult(
         raise ValueError("EventSet has no creator.")
     op = result.creator
 
-    serialized_op = serialization._serialize_operator(op)
-
-    nodes = {}
-    for node in op.inputs.values():
-        nodes[serialization._identifier(node)] = node
-    for node in op.outputs.values():
-        nodes[serialization._identifier(node)] = node
-
-    _ = serialization._unserialize_operator(serialized_op, nodes)
+    try:
+        serialized_op = serialization._serialize_operator(op)
+        nodes = {}
+        for node in op.inputs.values():
+            nodes[serialization._identifier(node)] = node
+        for node in op.outputs.values():
+            nodes[serialization._identifier(node)] = node
+
+        _ = serialization._unserialize_operator(serialized_op, nodes)
+    except ValueError as e:
```

Comment:
To be more fancy, I would create a boolean field "is_serializable" in the proto definition of the op: https://github.com/google/temporian/blob/main/temporian/proto/core.proto#L187

Then, we can have a utility function (like "has_leak") that check if a graph is serializable without relying on exceptions (since there might be other reasons for an exception).

Reply:
Would it make sense to make it a class variable instead? Then we could define it as true in the base operator and have the non-serializable ones override it to false. Or is there a reason to store this info in the proto itself?

Reply:
Having the definition in the proto means we can operator on the ops outside of the python language, which is a good option to have. This is why we have an operator def as a proto.